torch/_inductor/graph.py

```diff
@@ -277,7 +278,10 @@ def constant_name(self, name: str, device_override: torch.device):
         return alt_name
 
     def placeholder(self, target: str, args, kwargs):
-        example: torch.Tensor = super().placeholder(target, args, kwargs)
+        example = super().placeholder(target, args, kwargs)
+        if not torch.is_tensor(example):
+            self.graph_inputs_non_tensor[target] = example
+            example = torch.tensor(example)
```

Comment:
IIUC:
- Cannot just use a `TensorBox` because with those you cannot access original tensors
- Cannot do it earlier (in `compile_fx`) because (1) we need to know whether our argument is a non-tensor and (2) have access to `graph` dicts

Reply:
Cool, but I now remember that `torch.tensor(SymInt)` burns in the value, is that a problem?

Reply:
When I print `example` before converting while running `test_lowmem_dropout2_dynamic_shapes`, where non-tensors show up, I get 256. So it gets concretized earlier.

Reply:
Can you give an example of the output code to see if it actually dynamic?

Reply:
@jansel https://gist.github.com/nkaretnikov/6937a2251c53946580a8cafe85e72eaa

The `target` name of the non-tensor value is `add` and its value in `placeholder` is `256`.

In the output, it's defined in `__main__` but not used in `call`. (But I do use it in the lowering as `offset`.)

It appears in the output because we record it in `graph_inputs`. Otherwise, we would need to do additional filtering in `write_prefix` (where args unpacking is done).

`256` first appears in `CompiledFunction.forward` as part of `fw_outs`.

`offset`s without dyn shapes printed from the `philox_rand_like` lowering:
```
OFFSET 0
OFFSET 256
OFFSET 256
OFFSET 0
OFFSET 0
OFFSET 256
OFFSET 256
OFFSET 0
```

with dyn shapes:
```
OFFSET 0
OFFSET 32*s0
OFFSET 256
OFFSET 0
OFFSET 0
OFFSET 32*s0
OFFSET 256
OFFSET 0
```

So, is it fine? I'm not sure, but at least this looks consistent.

Reply:
I don't think this is correct. Forward looks good, with offsets in the kernels being 0 and 32*s0, but in the backward the `numel` https://gist.github.com/nkaretnikov/6937a2251c53946580a8cafe85e72eaa#file-cpu-txt-L141 and the offset (line 145) become hardcoded to 256. I don't know if this happens in aot autograd or somewhere in inductor, but either way, it's not dynamic.

Reply:
@ngimel I think this is due to https://github.com/pytorch/pytorch/pull/95461/files#r1127300079